examples/src/python/example/python_distribution/hello/superhello/BUILD

```diff
@@ -0,0 +1,58 @@
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+# Like Hello world, but built with a python_dist.
+# python_dist allows you to use setup.py to depend on C/C++ extensions.
+
+python_dist(
+  name='superhello',
+  sources=[
+    'super_greet.c',
+    'hello_package/hello.py',
+    'hello_package/__init__.py',
+    'setup.py'
+  ]
+)
+
+python_binary(
+  name='main',
+  source='main.py',
+  dependencies=[
+    ':superhello',
+  ]
+)
+
+# The targets below are for testing purposes only.
+# Their purpose is to expose an error that arises when a python_dist
+# contains a requirement that conflicts with the same requirement of
+# a different version in the consuming target. In this case, the pants
+# goal will fail and indicate that incomptible requirements exist.
+
+python_dist(
+  name='superhello_with_conflicting_dep',
+  sources=[
```

Comment:
This case was for declaring a dep in the binary below (3rdparty/pex which conflicts iirc), but I think this is all probably unnecessary because we know targets can handle this correctly already. 
So you are proposing we kill of dependencies field of python dist all together and have internal teams list their deps (like tensorflow) in install_requires instead? Because writing python code that depends on TF in your py_dist and linking to 3rdparty/python:tensorflow in the dependencies field of a consuming binary target seems confusing.